server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -95,32 +104,40 @@ public Task(
         this.startTime = startTime;
         this.startTimeNanos = startTimeNanos;
         this.headers = headers;
+        this.resourceStats = resourceStats;
     }
 
     /**
      * Build a version of the task status you can throw over the wire and back
      * to the user.
      *
-     * @param localNodeId
-     *            the id of the node this task is running on
-     * @param detailed
-     *            should the information include detailed, potentially slow to
-     *            generate data?
+     * @param localNodeId the id of the node this task is running on
+     * @param detailed    should the information include detailed, potentially slow to
+     *                    generate data?
+     * @param includeStats should the information include task resource stats
      */
-    public final TaskInfo taskInfo(String localNodeId, boolean detailed) {
+    public final TaskInfo taskInfo(String localNodeId, boolean detailed, boolean includeStats) {
```

Comment:
The reason behind introducing ```includeStats``` field was that,
1. The task index has strict mappings and adding additional fields **_might_** break backward compatibility. 
2. Some actions gets the task info with ```detailed: true``` like ```TransportRethrottleAction``` which might need description but not resource_info. 

But there is a recent PR which adds ```cancelled``` field to the mapping, so introducing new fields should not be a problem. Hence, remove ```includeStats``` field and failing back to detailed field for populating resource information. 

@sohami Do you think we should still have the includeStats for the [2] point or any other reason.

Reply:
Modified to use ```detailed``` flag to return resource consumption information.
Also, added private method that takes in ```includeStats``` to controls returning resource information. This was added to avoid storing resource information into task index as strict mappings are enforced and breaks when adding the ```resource_stats``` field. In the future, will remove this private method by adding resource stats into the task index.